Code/Framework/AzCore/AzCore/Component/Component.h

```diff
@@ -266,7 +266,7 @@ namespace AZ
                         _ComponentClass::RTTI_Type().ToString<AZStd::string>().c_str(), descriptor->GetName(), _ComponentClass::RTTI_TypeName());          \
                     return nullptr;                                                                                                     \
                 }                                                                                                                       \
-                else if (descriptor->GetName() != _ComponentClass::RTTI_TypeName())                                                     \
+                if (descriptor->GetName() != _ComponentClass::RTTI_TypeName())                                                     \
```

Comment:
What was wrong with this code being an `else if` before.
Based on the above if condition `if (strcmp(descriptor->GetName(), _ComponentClass::RTTI_TypeName()) != 0` I don't see how the `else if` conflicts with it.

Reply:
yeah, some of these were automatically fixed by a tool that I used to fix the virtual/override ones
In this case, the else is not necessary because of the return on line 267. 
I can revert it, its just that the tool has it as a "fix case" and I "fixed" the whole file